src/debtManagers/HookAMM.sol

```diff
@@ -0,0 +1,238 @@
+     * @return The current value of K.
+     */
+    function getK() public view returns (uint) {
+        uint duration = 7 days;
+        uint timeElapsed = block.timestamp - lastKUpdate;
+        if(timeElapsed > duration) {
+            return targetK;
+        }
+        uint targetWeight = timeElapsed;
+        uint prevWeight = duration - timeElapsed;
+        return (prevK * prevWeight + targetK * targetWeight) / duration;
+    }
+
+    /**
+     * @dev Calculates the DOLA reserve based on the current DBR reserve.
+     * @return The calculated DOLA reserve.
+     */
+    function getDolaReserve() public view returns (uint) {
+        return dola.balanceOf(address(this)) + getVirtualDolaReserves();
+    }
+
+    /**
+     * @dev Returns the current DBR reserve as the sum of dbr balance and claimable dbr
+     * @return The current DBR reserve.
+     */
+    function getDbrReserve() public view returns (uint) {
+        return getK() / getDolaReserve();
+    }
+
+    /**
+     * @dev Sets a new target K value.
+     * @param _K The new target K value.
+     */
+    function setTargetK(uint _K) external onlyGov {
+        require(_K > getDbrReserve(), "K must be larger than dbr reserve");
+        prevK = getK();
+        targetK = _K;
+        lastKUpdate = block.timestamp;
+        emit SetTargetK(_K);
+    }
+
+    function setMaxDolaPrice(uint _maxDolaPrice) external onlyGov {
+        require(_maxDolaPrice >= minDolaPrice, "Max DOLA price must be higher or equal to min DOLA price");
+        maxDolaPrice = _maxDolaPrice;
+    }
+
+    function setMinDolaPrice(uint _minDolaPrice) external onlyGov {
+        require(maxDolaPrice >= _minDolaPrice, "Max DOLA price must be higher or equal to min DOLA price");
+        minDolaPrice = _minDolaPrice;
+    }
+
+    function getVirtualDolaReserves() internal view returns(uint) {
+        return sqrt(1e18 * getK() / minDolaPrice);
+    }
+
+    /**
+     * @dev Allows users to buy DBR with DOLA.
+     * WARNING: Never expose this directly to a UI as it's likely to cause a loss unless a transaction is executed immediately.
+     * Instead use the sDolaHelper function or custom smart contract code.
+     * @param exactDolaIn The exact amount of DOLA to spend.
+     * @param exactDbrOut The exact amount of DBR to receive.
+     * @param to The address that will receive the DBR.
+     */
+    function buyDBR(uint exactDolaIn, uint exactDbrOut, address to) external buyHook {
+        require(to != address(0), "Zero address");
+        _invariantCheck(exactDolaIn, exactDbrOut, getDbrReserve());
+        dola.transferFrom(msg.sender, address(this), exactDolaIn);
+        uint dbrBal = dbr.balanceOf(address(this));
+        if(exactDbrOut > dbrBal)
+            dbr.mint(address(this), exactDbrOut - dbrBal);
+        uint exactDbrOutAfterFee = exactDbrOut * (1e18 - dbrBuyFee) / 1e18;
+        feesAccrued += exactDbrOut - exactDbrOutAfterFee;
```

Comment:
Why mint the fee at all? Instead we should deduct it from the minted amount (or burn it from balance) and remove the feeRecipient role entirely.